server/command.go

```diff
@@ -21,70 +21,68 @@ const helpText = "###### Mattermost Jira Plugin - Slash Command Help\n" +
 	"  * `delete <number or URL>` - Delete a known instance, select the first remaining as the current\n" +
 	""
 
-func getCommand() *model.Command {
-	return &model.Command{
-		Trigger:          "jira",
-		DisplayName:      "Jira",
-		Description:      "Integration with Jira.",
-		AutoComplete:     true,
-		AutoCompleteDesc: "Available commands: connect, disconnect, help",
-		AutoCompleteHint: "[command]",
+type CommandHandlerFunc func(p *Plugin, c *plugin.Context, args ...string) *model.CommandResponse
+
+type CommandHandler struct {
+	handlers       map[string]CommandHandlerFunc
+	defaultHandler CommandHandlerFunc
+}
+
+var jiraCommandHandler = CommandHandler{
+	handlers: map[string]CommandHandlerFunc{
+		"instance/add/server": executeInstanceAddServer,
+		"instance/add/cloud":  executeInstanceAddCloud,
+		"instance/list":       executeInstanceList,
+		"instance/select":     executeInstanceSelect,
+		"instance/delete":     executeInstanceDelete,
+		"connect":             executeConnect,
+		"disconnect":          executeDisconnect,
+	},
+	defaultHandler: commandHelp,
+}
+
+func (ch CommandHandler) Handle(p *Plugin, c *plugin.Context, args ...string) *model.CommandResponse {
+	for n := len(args); n > 0; n-- {
+		h := ch.handlers[strings.Join(args[:n], "/")]
+		if h != nil {
```

Comment:
nit: more idiomatic way:
```
if h, ok := ch.handlers[strings.Join(args[:n], "/")]; ok {
   return h(p, c, args[n:]...)
}
```
But 0/5 personally (because there's no chance you'll have a nil value instead of a missing entry).

Reply:
I disagree. In this case there is no semantic difference between "no entry" and a `nil` entry, so checking the value is the _right way_.

Reply:
Sounds good.